src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -27,6 +27,36 @@ public class PackagesConfigWriter : IDisposable
         /// <summary>
         /// Create a packages.config writer
         /// </summary>
+        /// <param name="fullPath">the full path to write the XML packages.config file into, or load existing packages.config from</param>
+        /// <param name="createNew">Whether to create a new packages.config file, or load an existing one</param>
+        public PackagesConfigWriter(string fullPath, bool createNew)
+            : this(DefaultFrameworkNameProvider.Instance, fullPath, createNew)
+        {
+        }
+
+        public PackagesConfigWriter(IFrameworkNameProvider frameworkMappings, string fullPath, bool createNew)
+        {
+            _closed = false;
+            _frameworkMappings = frameworkMappings;
+
+            if (createNew)
+            {
+                _xDocument = new XDocument();
+                EnsurePackagesNode();
```

Comment:
@danliu you want to explain why it shouldn't be at a higher level. I assume this is because it adds the packages node, to a new XML, but if it is not there it is considered a corrupted scenario?

What happens in that case, do we get a reasonable error message?